File: 3351-maximize-happiness-of-selected-children/solution.java
import java.util.Arrays;

class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {

        Arrays.sort(happiness);
        
        long total = 0;
        int n = happiness.length;

        for (int i = 0; i < k; i++) {

            int current = happiness[n - 1 - i] - i;

            if (current <= 0) break;
            
            total += current;
        }
        
        return total;
    }
}

File: 3351-maximize-happiness-of-selected-children/solution.ts
function maximumHappinessSum(happiness: number[], k: number): number {
    happiness.sort((a, b) => b - a);

    let total = 0;
    
    for (let i = 0; i < k; i++) {
        const val = happiness[i] - i;
        
        if (val <= 0) break;
        
        total += val;
    }

    return total;
};
